rust/sdk-processor/src/steps/common/parquet_buffer_step.rs

```diff
@@ -0,0 +1,387 @@
+    ///
+    /// We check the size of the buffer + the size of the incoming data before appending it.
+    /// If the sum of the two exceeds the maximum limit size, it uploads the buffer content to GCS to avoid
+    /// spliting the batch data, allowing for more efficient and simpler version tracking.
+    async fn handle_buffer_append(
+        &mut self,
+        parquet_type: ParquetTypeEnum,
+        parquet_data: ParquetTypeStructs,
+        cur_batch_metadata: &TransactionMetadata,
+        upload_metadata_map: &mut HashMap<ParquetTypeEnum, TransactionMetadata>,
+    ) -> Result<(), ProcessorError> {
+        let mut file_uploaded = false;
+
+        // Get or initialize the buffer for the specific ParquetTypeEnum
+        let buffer = self
+            .internal_buffers
+            .entry(parquet_type)
+            .or_insert_with(|| {
+                debug!(
+                    "Initializing buffer for ParquetTypeEnum: {:?}",
+                    parquet_type,
+                );
+                ParquetBuffer::new(&parquet_type)
+            });
+
+        let curr_batch_size_bytes = parquet_data.calculate_size();
+
+        debug!(
+            "Current batch size for {:?}: {} bytes, buffer size before append: {} bytes",
+            parquet_type, curr_batch_size_bytes, buffer.buffer_size_bytes,
+        );
+
+        // If the current buffer size + new batch exceeds max size, upload the buffer
+        if buffer.buffer_size_bytes + curr_batch_size_bytes > self.buffer_max_size {
+            println!(
+                "Buffer size {} + batch size {} exceeds max size {}. Uploading buffer for {:?}.",
+                buffer.buffer_size_bytes, curr_batch_size_bytes, self.buffer_max_size, parquet_type
+            );
+
+            // Take the current buffer to upload and reset the buffer in place
+            let struct_buffer = std::mem::replace(
+                &mut buffer.buffer,
+                ParquetTypeStructs::default_for_type(&parquet_type),
+            );
+            self.buffer_uploader.handle_buffer(struct_buffer).await?;
+            file_uploaded = true;
+
+            // update this metadata before insert
+            upload_metadata_map
+                .insert(parquet_type, buffer.current_batch_metadata.clone().unwrap());
+            buffer.buffer_size_bytes = 0;
+        }
+
+        // Append new data to the buffer
+        Self::append_to_buffer(buffer, parquet_data)?;
+        buffer.update_current_batch_metadata(cur_batch_metadata);
+
+        // if it wasn't uploaded -> we update only end_version, size, and last timestamp
+        if file_uploaded {
+            if let Some(buffer_metadata) = &mut buffer.current_batch_metadata {
+                buffer_metadata.start_version = cur_batch_metadata.start_version;
+                buffer_metadata.start_transaction_timestamp =
+                    cur_batch_metadata.start_transaction_timestamp.clone();
+            }
+        }
```

Comment:
Is this still necessary if we already did line 147?

Reply:
yeah, it's slightly different b/c here we are updating the start_version  and start_trxn_timestamp, which only needs to be updated after we upload the file

Reply:
WOuldn't 2nd part of the function (lines 45-46) take care of updating after uploading the file too?

Reply:
oh right, if we set `none` after upload, it will be handled when we actually handle the current batch.